stl/inc/shared_mutex

```diff
@@ -225,50 +286,100 @@ class shared_lock { // shareable lock
 public:
     using mutex_type = _Mutex;
 
+    _Post_satisfies_(_Owns == false)
+    _Post_satisfies_(_Pmtx == nullptr)
     shared_lock() noexcept = default;

```

Comment:
I broke the tests here (sorry!). In `VSO_0000000_instantiate_iterators_misc` we're getting:

```
shared_mutex(291) : warning C28251: Inconsistent annotation for '{ctor}': this instance has no annotations. See d:\build\out\inc\shared_mutex(291).
```

I am not immediately sure what's happening here - it sure looks like an `/analyze` compiler bug.

Reply:
Interesting!
Guess this is not so common to annotate a defaulted constructor.  

Unfortunately this week probably I won’t be able to take a look on it. Next week should be less busy.